cmd/janitor/gcp_janitor.py

```diff
@@ -76,6 +76,12 @@
         Resource('', 'logging', 'sinks', None, None, None, False, False, ['_Default', '_Required']),
     ],
 
+    # pubsub resources
+    'pubsub.googleapis.com': [
+        Resource('', 'pubsub', 'subscriptions', None, None, None, True, True, None),
+        Resource('', 'pubsub', 'topics', None, None, None, True, True, None)
```

Comment:
are there certain topics we shouldn't delete? looking at a random sampling of projects, I see some with topics like
- container-analysis-notes-v1
- container-analysis-notes-v1beta1
- container-analysis-occurrences-v1
- container-analysis-occurrences-v1beta1

I don't really understand why there are container analysis pubsub topics in boskos test projects, but I also don't understand if there will be any issues if we delete them.

Reply:
Hmm I see, I did some search and found https://cloud.google.com/container-registry/docs/pub-sub-notifications#create_pubsub_topics, it looks they are automatically created after enabling the Container Analysis API, so we cannot simply ask Janitor to clean them up.

Also it looks less ideal to me to add them as `preserved_names`, since we don't know if these are all the pubsub resources we need to preserve, and we don't know if someday their names will be changed.

Considering it from another angle, would it make sense to make Janitor more customizable? Since different projects might have different requirements on what resources they want to clean up / preserve, we can by default get Janitor cleaning up some heavy-weight resources (like GKE and GCE), but also allowing users to configure additional resources they want to clean up?

Reply:
I'm tending to leave these topics uncleaned for safety reasons, so I added them to `preserved_names`.